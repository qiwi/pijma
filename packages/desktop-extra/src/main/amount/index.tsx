import { formatMoney } from '@qiwi/common-formatters'
import React from 'react'

import AmountProps from './AmountProps'

export const Amount = ({
  value,
  currency,
  symbol,
  sign,
  fractionLength = 2,
  digitDelimiter = ' ',
  fractionDelimiter,
}: AmountProps) => {
  return (
    <span>
      {formatMoney(value, {
        currencyCode: currency,
        currencySymbol: symbol,
        sign,
        fractionLength,
        digitDelimiter,
        fractionDelimiter,
      })}
    </span>
  )
}
