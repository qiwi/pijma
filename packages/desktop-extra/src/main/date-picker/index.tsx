import { Icon, Overlay, styled } from '@qiwi/pijma-core'
import { MaskTextField } from '@qiwi/pijma-desktop'
import { format } from 'date-fns'
import React, { Component } from 'react'
import { DayPicker } from 'react-day-picker'
import { WithTranslation, withTranslation } from 'react-i18next'

import { COLOR } from '../theme'
import DatePickerProps from './DatePickerProps'
import DatePickerState from './DatePickerState'
import { locales, TLanguage } from './locale'
import Wrap from './wrap'
import 'react-day-picker/dist/style.css'

const PickerDropdown = styled('div')`
  position: absolute;
  background: #fff;
  z-index: 10;
  box-shadow: ${COLOR.SHADOW.Z3};
  border-radius: 8px;

  .Selectable {
    --rdp-cell-size: 40px;
    --rdp-accent-color: rgba(255, 140, 0, 0.4);
    --rdp-background-color: rgba(255, 140, 0.1);
    /* Switch to dark colors for dark themes */
    --rdp-accent-color-dark: rgba(255, 140, 0);
    --rdp-background-color-dark: rgba(255, 140, 0.7);
    /* Outline border for focused elements */
    --rdp-outline: 2px solid var(--rdp-accent-color);
    /* Outline border for focused and selected elements */
    --rdp-outline-selected: 2px solid rgba(0, 0, 0, 0.75);

    .my-selected {
      padding: 0px;
      width: 40px;
      height: 40px;
      color: black;
      background-color: rgba(255, 140, 0);
      border-radius: 50%;
    }

    .my-head-cell {
      width: 40px;
      height: 40px;
      padding: 4px;
    }

    .my-month {
      margin: 1rem;
    }

    .my-caption {
      margin-bottom: 10px;
      margin-left: 8px;
      display: flex;
      justify-content: space-between;
    }

    .my-day {
      width: 40px;
      height: 40px;
      padding: 4px;
    }

    .my-today {
      color: rgb(208, 2, 27);
    }

    .my-today {
      color: rgb(208, 2, 27);
    }

    .my-caption-label {
      font-size: 20px;
    }
  }
`

const Container = styled('div')`
  position: relative;
  width: 100%;
`

class _DatePicker extends Component<
  DatePickerProps & WithTranslation,
  DatePickerState
> {
  myRef: any
  _timeoutID: number

  constructor(props: DatePickerProps & WithTranslation) {
    super(props)
    this.handleDayClick = this.handleDayClick.bind(this)
    this.state = {
      value: '',
      date: undefined,
      showPicker: false,
      errorDate: false,
      dateRangeMask: [/\d/, /\d/, '.', /\d/, /\d/, '.', /\d/, /\d/, /\d/, /\d/],
    }
    this.myRef = React.createRef()
    this._timeoutID = -1
  }

  handleDayClick(date: Date) {
    this.setState({ date, value: format(date, 'dd.MM.yyyy') })
    if (this.props.onChange) {
      this.props.onChange(date)
    }
  }

  onFocus() {
    clearTimeout(this._timeoutID)
    this.setState({ showPicker: true })
  }

  onBlur() {
    this._timeoutID = window.setTimeout(() => {
      this.setState({ showPicker: false })
    })
  }

  onChange(value: string) {
    const date = this.parseDate(value)
    if (date) {
      this.setState({ errorDate: false })
    } else {
      this.setState({ errorDate: true })
    }

    this.setState({ value, date })

    if (this.props.onChange) {
      this.props.onChange(date)
    }
  }

  parseDate(date: string) {
    const pattern = /^(0?[1-9]|[12]\d|3[01])\.(0?[1-9]|1[0-2])\.\d{4}$/
    if (pattern.test(date)) {
      const parseDate = date.split('.')
      return new Date(`${parseDate[1]}, ${parseDate[0]}, ${parseDate[2]}`)
    }
  }

  render() {
    const { t, i18n } = this.props

    return (
      <Wrap onFocus={this.onFocus.bind(this)} onBlur={this.onBlur.bind(this)}>
        <MaskTextField
          type="text"
          title={this.props.title || t('datePicker:date')}
          hint={<Icon name="calendar" />}
          onChange={this.onChange.bind(this)}
          value={this.state.value}
          error={this.state.errorDate ? t('datePicker:wrongDate') : undefined}
          mask={this.state.dateRangeMask}
        />
        <Container ref={this.myRef} />

        <Overlay
          show={this.state.showPicker}
          onHide={() => {
            /* noop */
          }}
          placement="bottom"
          container={this.myRef.current}
          target={this.myRef.current}
        >
          {() => (
            <PickerDropdown onFocus={this.onFocus.bind(this)}>
              <DayPicker
                className="Selectable"
                onDayClick={this.handleDayClick.bind(this)}
                selected={this.state.date}
                month={this.state.date}
                locale={locales[i18n.language as TLanguage]}
                modifiersClassNames={{
                  selected: 'my-selected',
                  today: 'my-today',
                }}
                classNames={{
                  day: 'my-day',
                  head_cell: 'my-head-cell',
                  caption: 'my-caption',
                  month: 'my-month',
                }}
              />
            </PickerDropdown>
          )}
        </Overlay>
      </Wrap>
    )
  }
}

const DatePicker = withTranslation()(_DatePicker)

export { DatePicker }
