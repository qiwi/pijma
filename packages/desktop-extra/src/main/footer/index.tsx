import { Flex, styled } from '@qiwi/pijma-core'
import React from 'react'
import { useTranslation } from 'react-i18next'

import Spreader from '../spreader'

const Wrapper = styled(Spreader)`
  margin-top: 75px;
  color: #4a4a4a;
  margin-bottom: 50px;
  color: #4a4a4a;
`

const References = styled('a')`
  cursor: pointer;
  text-decoration: none;
  color: #4a4a4a;
  :hover {
    color: ${(props) => props.theme.color.gray.dark};
  }
`

interface FooterProps {
  email: string
}

function setLang(lng: string) {
  localStorage.setItem('i18nextLng', lng)
}

const Footer = ({ email }: FooterProps) => {
  const { t, i18n } = useTranslation()
  const lang = i18n.language.replace(/(\w+).*/, '$1')
  return (
    <Wrapper>
      <Flex p={0} m={0} justify="space-between">
        <div>
          © {new Date().getFullYear()}, {t('footer:qiwi')} № 2241
        </div>
        <References href={email || '/#'}>{t('footer:support')}</References>
      </Flex>
      <Flex p={0} m={0} justify="space-between">
        <span>&nbsp;</span>
        <span>
          {lang === 'ru' ? (
            <span>РУС</span>
          ) : (
            <a
              style={{ paddingLeft: 0 }}
              id="switch-ru"
              href={location.href}
              onClick={() => setLang('ru')}
            >
              РУС
            </a>
          )}{' '}
          /{' '}
          {lang === 'en' ? (
            <span>EN</span>
          ) : (
            <a
              style={{ paddingLeft: 0 }}
              id="switch-en"
              href={location.href}
              onClick={() => setLang('en')}
            >
              EN
            </a>
          )}
        </span>
      </Flex>
    </Wrapper>
  )
}

export { Footer }
