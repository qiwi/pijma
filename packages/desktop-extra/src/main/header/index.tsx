import { Flex, FlexItem, styled } from '@qiwi/pijma-core'
import { HeaderMenu } from '@qiwi/pijma-desktop'
import React, { Component } from 'react'

import { Logo } from '../logo'
import { Product } from '../product'
import { B2bSpinner } from '../spinner'
import { COLOR, DIMEN } from '../theme'
import { UserProfile } from '../user-profile'
import HeaderProps from './HeaderProps'

const HeaderContainer = styled('header')`
  width: 100%;
  position: relative;
  top: 0;
  left: 0;
  height: 80px;
  padding: 0;
  margin: 0 auto;
  color: ${COLOR.TEXT.Secondary2};
  background: ${COLOR.BACKGROUND.Primary};
  font-size: 11pt;
`

const HeaderContent = styled('div')`
  width: ${DIMEN.SITE_WIDTH};
  margin: 0 auto;
  height: 100%;
`

export class Header extends Component<HeaderProps> {
  render() {
    const {
      isLoading,
      title,
      selectorData,
      navItems,
      userName,
      onItemChange,
      onLogout,
    } = this.props

    const menuItems = navItems.map(({ name, path }) => {
      const href = typeof path === 'string' ? path : path.value
      return {
        href,
        title: name,
        active: location.pathname === path,
        onClick: () => {
          window.location.href = href
        },
      }
    })

    return (
      <HeaderContainer>
        <HeaderContent>
          {isLoading ? (
            <Flex justify="center" align="center" css={{ height: '100%' }}>
              <B2bSpinner />
            </Flex>
          ) : (
            <Flex p={0} m={0} justify="space-between" align="center">
              <Flex align="center">
                <FlexItem mr={10}>
                  {/* TODO: move to nav controller */}
                  <a
                    href={`http://${window.location.host}/service/core/merchants`}
                  >
                    <Logo />
                  </a>
                </FlexItem>
                <Product
                  title={title}
                  selectorData={selectorData}
                  onChange={onItemChange}
                />
                <FlexItem ml={10} p={0} height={20}>
                  <HeaderMenu children={menuItems} />
                </FlexItem>
              </Flex>
              <FlexItem m={0} p={0}>
                <UserProfile onClick={onLogout} userName={userName} />
              </FlexItem>
            </Flex>
          )}
        </HeaderContent>
      </HeaderContainer>
    )
  }
}
