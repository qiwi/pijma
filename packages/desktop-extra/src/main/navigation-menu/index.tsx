import { styled } from '@qiwi/pijma-core'
import React, { Component } from 'react'

import { Menu } from '../menu'
import NavigationMenuProps from './NavigationMenuProps'

const ItemWrap = styled('div')`
  padding: 28px 0 24px;
  font-size: 15px;
`

export class NavigationMenu extends Component<NavigationMenuProps> {
  render() {
    const Link = this.props.link
    const { items, isItemSelected, onChange } = this.props

    const menuItems = items.map(
      ({ name, path, node: _node, value: _value }) => {
        let node = _node
        let value = _value
        if (path) {
          node = (
            <Link to={path}>
              <ItemWrap>{name}</ItemWrap>
            </Link>
          )
          value = path
        }
        return { name, value, node }
      },
    )

    return (
      <Menu
        default={this.props.active}
        items={menuItems}
        isItemSelected={isItemSelected}
        onChange={onChange}
      />
    )
  }
}
