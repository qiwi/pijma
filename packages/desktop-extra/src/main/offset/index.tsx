import { styled } from '@qiwi/pijma-core'

import OffsetProps from './OffsetProps'

export const Offset = styled('div')<OffsetProps>`
  margin-top: ${(props: OffsetProps) => props.mt || 0}px;
  margin-right: ${(props: OffsetProps) =>
    props.mr === undefined ? 40 : props.mr}px;
  margin-bottom: ${(props: OffsetProps) => props.mb || 0}px;
  margin-left: ${(props: OffsetProps) =>
    props.ml === undefined ? 40 : props.ml}px;
`
