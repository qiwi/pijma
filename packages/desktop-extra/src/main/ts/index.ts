export * from '@qiwi/pijma-core'

export * from './amount'
export * from './divider'
export * from './dropdown'
export * from './logo'
export * from './obfuscatedPan'
export * from './offset'
export * from './spinner'
export * from './spreader'
export * from './step'
export * from './trinary-check-icon'
export * from './checkbox'
export * from './chip'
export * from './date-picker'
export * from './date-range-picker'
export * from './error-message'
export * from './file-uploader'
export * from './footer'
export * from './icon'
export * from './menu'
export * from './navigation-menu'
export * from './paginator'
export * from './product'
export * from './select'
export * from './stepper'
export * from './user-profile'
export * from './header'
export * from './empty-page'
export * from './table'
