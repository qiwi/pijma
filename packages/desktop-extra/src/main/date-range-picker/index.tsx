import { Icon, Overlay, styled } from '@qiwi/pijma-core'
import { TextField } from '@qiwi/pijma-desktop'
import { format } from 'date-fns'
import React, { Component } from 'react'
import { DayPicker, addToRange } from 'react-day-picker'
import { WithTranslation, withTranslation } from 'react-i18next'

import Wrap from '../date-picker/wrap'
import { COLOR } from '../theme'
import DateRangePickerState from './DateRangePickerState'
import DateRangerPickerProps from './DateRangerPickerProps'
import { locales, TLanguage } from '../date-picker/locale'

export const PickerDropdown = styled('div')`
  position: absolute;
  background: #fff;
  z-index: 10;
  border-radius: 8px;
  box-shadow: ${COLOR.SHADOW.Z3};

  .Selectable {
    .my-selected {
      padding: 0px;
      width: 40px;
      height: 40px;
      color: black;
      background-color: rgba(255, 140, 0, 0.4);
      border-radius: unset;
    }

    .my-range-start {
      background-color: rgb(255, 140, 0);
      border-radius: 50% 0px 0px 50%;
    }

    .my-range-middle {
    }

    .my-range-end {
      background-color: rgb(255, 140, 0);
      border-radius: 0px 50% 50% 0px;
    }

    .my-head-cell {
      width: 40px;
      height: 40px;
      padding: 4px;
    }

    .my-caption {
      margin-bottom: 10px;
      margin-left: 8px;
    }

    .my-month {
      margin: 1rem;
    }

    .my-months {
      margin: 0 12px;
      display: flex;
      position: relative;
    }

    .my-day {
      width: 40px;
      height: 40px;
      padding: 4px;
    }

    .my-nav {
      position: absolute;
      translateY(-50%);
      top: 50%;
    }

    .my-caption-start {
      .my-nav {
        left: -6px;
      }
    }

    .my-caption-end {
      .my-nav {
        right: -6px;
      }
    }

    .my-day.my-selected.my-range-end.my-range-start {
      border-radius: 50%;
    }

    .my-today {
      color: rgb(208, 2, 27);
    }

    .my-caption-label {
      font-size: 20px;
    }
  }
`

const Container = styled('div')`
  position: relative;
  width: 100%;
`

const RedHelpText = styled('div')`
  color: ${COLOR.TEXT.Error};
`

const InputWrapper = styled('div')`
  input {
    color: transparent;
    text-shadow: 0 0 0 #000;
  }
`

class DisabledInputDateRangePicker extends Component<
  DateRangerPickerProps & WithTranslation,
  DateRangePickerState
> {
  static defaultProps = {
    numberOfMonths: 2,
  }

  myRef: any
  inputRef: any

  constructor(props: DateRangerPickerProps & WithTranslation) {
    super(props)
    this.state = {
      showPicker: false,
      errorDate: false,
      helpText: null,
    }
    this.myRef = React.createRef()
    this.inputRef = React.createRef()
    this._timeoutID = -1
  }

  _timeoutID: number

  formatPeriod = (dateFrom: Date, dateTo: Date) => {
    if (!dateFrom) {
      return ''
    }

    if (
      !dateTo ||
      format(dateFrom, 'dd.MM.yyyy') === format(dateTo, 'dd.MM.yyyy')
    ) {
      return format(dateFrom, 'dd.MM.yyyy')
    }

    return `${format(dateFrom, 'dd.MM.yyyy')} - ${format(dateTo, 'dd.MM.yyyy')}`
  }

  handleDayClick = (date: any, { disabled }: any) => {
    if (disabled) {
      return
    }

    const normalizeDate = {
      from: this.props.dateFrom,
      to: this.props.dateTo || this.props.dateFrom,
    }

    const range = addToRange(date, normalizeDate)
    const { from: dateFrom, to: dateTo } = range || normalizeDate

    if ((dateFrom || dateTo) && this.props.onChange) {
      this.props.onChange({ dateFrom, dateTo: dateTo || dateFrom })
    }
  }

  onFocus = () => {
    clearTimeout(this._timeoutID)
    this.setState({ showPicker: true })
  }

  onBlur = () => {
    this._timeoutID = window.setTimeout(() => {
      this.setState({ showPicker: false })
    })
  }

  onIconClick = () => {
    this.inputRef.current.focus()
  }

  render() {
    const { t, i18n } = this.props

    const { errorDate, helpText, showPicker } = this.state
    const {
      minDate,
      maxDate,
      dayPickerProps,
      title,
      numberOfMonths,
      dateTo,
      dateFrom,
      inputValue,
    } = this.props
    const modifiers = { start: dateFrom, end: dateTo }
    const disabled: any = {}
    const selectedDays = { from: dateFrom, to: dateTo }
    const value = inputValue || this.formatPeriod(dateFrom, dateTo)

    if (minDate) {
      disabled.before = minDate
    }
    if (maxDate) {
      disabled.after = maxDate
    }

    return (
      <Wrap
        onFocus={this.onFocus}
        onBlur={this.onBlur}
        ref={this.inputRef}
        tabIndex={0}
      >
        <InputWrapper>
          <TextField
            type="text"
            title={title || t('datePicker:date')}
            hint={
              <div onClick={this.onIconClick}>
                <Icon name="calendar" />
              </div>
            }
            onChange={() => {
              /* noop */
            }}
            value={value}
            error={errorDate ? t('datePicker:wrongDate') : undefined}
            help={<RedHelpText>{helpText}</RedHelpText>}
          />
        </InputWrapper>
        <Container ref={this.myRef} />

        <Overlay
          show={showPicker}
          placement="bottom"
          container={this.myRef.current}
          target={this.myRef.current}
        >
          {() => (
            <PickerDropdown onFocus={this.onFocus}>
              <DayPicker
                mode={'range'}
                className="Selectable"
                disabled={disabled}
                locale={locales[i18n.language as TLanguage]}
                numberOfMonths={numberOfMonths}
                selected={selectedDays}
                modifiers={modifiers}
                onDayClick={this.handleDayClick}
                modifiersClassNames={{
                  selected: 'my-selected',
                  today: 'my-today',
                  range_start: 'my-range-start',
                  range_middle: 'my-range-middle',
                  range_end: 'my-range-end',
                }}
                classNames={{
                  day: 'my-day',
                  head_cell: 'my-head-cell',
                  caption: 'my-caption',
                  month: 'my-month',
                  months: 'my-months',
                  caption_start: 'my-caption-start',
                  caption_end: 'my-caption-end',
                  nav: 'my-nav',
                  caption_label: 'my-caption-label',
                }}
                {...dayPickerProps}
              />
            </PickerDropdown>
          )}
        </Overlay>
      </Wrap>
    )
  }
}

const DateRangePicker = withTranslation()(DisabledInputDateRangePicker)

export { DateRangePicker }
