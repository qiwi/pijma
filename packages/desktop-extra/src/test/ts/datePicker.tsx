import { ThemeProvider, themes } from '@qiwi/pijma-core'
import React from 'react'
import renderer from 'react-test-renderer'

import { DatePicker } from '../../main/ts'

describe('DatePicker', () => {
  it('is properly exported from lib index', () => {
    expect(DatePicker).not.toBeUndefined()
  })

  it('renders correctly', () => {
    const datePicker = renderer
      .create(
        <ThemeProvider theme={themes.orange}>
          <DatePicker title="Date" />
        </ThemeProvider>,
        {
          createNodeMock: (element) => {
            if (element.type === 'input') {
              // mock a focus function
              return {
                focus: () => {
                  /* noop */
                },
              }
            }
            return null
          },
        },
      )
      .toJSON()
    expect(datePicker).toMatchSnapshot()
  })
})
