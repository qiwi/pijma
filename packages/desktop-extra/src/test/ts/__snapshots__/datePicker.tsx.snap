// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker renders correctly 1`] = `
<div
  className="css-1b7d9rb"
  onBlur={[Function]}
  onFocus={[Function]}
>
  <div
    className="css-1haykwu"
  >
    <div
      className="css-1hy2az2"
    >
      <div
        className="css-1mw5xg7"
      >
        <label
          className="css-195rf93"
        >
          Date
        </label>
      </div>
    </div>
    <div
      className="css-1bx5ylf"
    >
      <div
        className="css-70qvj9"
      >
        <input
          autoComplete="off"
          className="css-1haa77m"
          defaultValue=""
          disabled={false}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          type="text"
        />
        <div
          className="css-t03j0v"
        >
          <svg
            className="css-9uy14h"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              className="css-0"
              d="M17 4h1.006A2.995 2.995 0 0 1 21 7.01v9.98A3.004 3.004 0 0 1 18.006 20H5.994A2.995 2.995 0 0 1 3 16.99V7.01A3.004 3.004 0 0 1 5.994 4H7V3a1 1 0 1 1 2 0v1h6V3a1 1 0 0 1 2 0v1zm0 2v1a1 1 0 0 1-2 0V6H9v1a1 1 0 1 1-2 0V6H5.994C5.45 6 5 6.45 5 7.01v9.98c0 .566.443 1.01.994 1.01h12.012c.545 0 .994-.45.994-1.01V7.01A.995.995 0 0 0 18.006 6H17zm-9 6a1 1 0 1 1 0-2 1 1 0 0 1 0 2zm4 0a1 1 0 1 1 0-2 1 1 0 0 1 0 2zm4 0a1 1 0 1 1 0-2 1 1 0 0 1 0 2zm-8 4a1 1 0 1 1 0-2 1 1 0 0 1 0 2zm4 0a1 1 0 1 1 0-2 1 1 0 0 1 0 2zm4 0a1 1 0 1 1 0-2 1 1 0 0 1 0 2z"
              fill="#000"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      className="css-1abbrc"
    />
  </div>
  <div
    className="css-1xm32e0"
  />
</div>
`;
